my-dsl.md

```diff
@@ -1,38 +1,79 @@
 
+The language is an internal, fragmentary language, so its computational model
+is inherited from its host language Javascript. This means that when tests which
+use should.js run, the should.js code is interpreted as valid Javascript code,
+often intermixed with general Javascript, and is fed to the JS interpreter to be
+executed.
 
 # DSL-ness
-_Fowler writes about a spectrum of languages, from general-purpose languages to 
-"purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
-and why?_ 
+_Fowler writes about a spectrum of languages, from general-purpose languages to
+"purely" domain-specific. Where does the DSL you chose fall on this spectrum,
+and why?_
 
+should.js is purely domain-specific. Its lack of Turing completeness is the
+biggest indicator: the langauge does not have variables, conditionals, or
+looping constructs (although all of these may be found interspersed with
+should.js code in native Javascript, due to the fragmentary nature of the DSL).
+
+The only thing should.js is useful for is making assertions about code in a
+self-documenting way.
 
 # Internal or external?
-_Is the language implemented as an internal or external DSL? 
+_Is the language implemented as an internal or external DSL?
 Justify your answer._
 
+It is internal: should.js code is valid Javascript.
 
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
+Javascript.
 
 # Benefits
-_Identify one potential benefit of the DSL: how is a programmer's life or a 
+_Identify one potential benefit of the DSL: how is a programmer's life or a
 company's bottom line made easier by the existence of this language?_
 
+should.js allows the programmer to write unit tests that not only prove the
+correctness of the program, but also document what the correct behavior should
+be. This is done in a way that is readable even if the reader does not know
+anything about the host language. For example,
+
+```javascript
+this.obj.should.have.property('id').which.is.a.Number()
+this.obj.should.have.property('path')
+```
+
+should be understandable so long as the reader knows what 'id' and 'path' refer
+to (that is, the reader is familiar with the domain). This makes the language
+useful as a tool for communicating with domain experts, one of the benefits of a
+DSL which Fowler mentions in chapter 2.
 
 # Drawbacks
-_Identify one potential drawback of the DSL: what does a programmer or company 
+_Identify one potential drawback of the DSL: what does a programmer or company
 lose by using this DSL instead of a general-purpose language?_
+
+The biggest drawback of should.js may be, ironically, the same as its biggest
+strength. The syntax is very close to natural language. This is what makes the
+code so reasonable, but it is also what Fowler warns against on page 42. The
+large amount of syntactic sugar needed to make the language read this way can
+also make it difficult for the programmer to learn, which could slow down
```

Comment:
```python
import this
```

It's definitely hard when there isn't an obvious reason to prefer one method or another, and this is where things like coding standards and style guides come into play. Anything a language doesn't enforce should be enforced by your organization (even if it's just you doing work on your own).  It is definitely debatable, however, as to whether or not the language **should** enforce this.  In this case I think being able to use many different words in order to make it sound more like natural language is worthwhile given the intent, but I agree that it's hard to use.